src/components/YourCompany/BusinessContactInfo.vue

```diff
@@ -1,293 +1,77 @@
-import { mask } from 'vue-the-mask'
+import { Component, Emit, Mixins } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
 
-// Interfaces & enums
-import { BusinessContactIF } from '@/interfaces'
+// Interfaces
+import {
+  ActionBindingIF,
+  BusinessContactIF,
+  BusinessSnapshotIF,
+  IncorporationFilingIF
+} from '@/interfaces'
 
 // Mixins
-import { CommonMixin } from '@/mixins'
-
-@Component({
-  directives: { mask }
-})
-export default class BusinessContactInfo extends Mixins(CommonMixin) {
-  @Prop()
-  private businessContact!: BusinessContactIF
-
-  @Prop()
-  private originalBusinessContact!: BusinessContactIF
-
-  // Properties
-  private isEditing: boolean = false
-  private contact: BusinessContactIF
-  private formValid: boolean = false
-
-  // Rules
-  private emailRules = [
-    (v: string) => !!v || 'Email address is required',
-    (v: string) => {
-      const pattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
-      return pattern.test(v) || 'Valid email is required'
-    }
-  ]
-
-  private phoneRules = [
-    (v: any) => !v || (v.length === 0 || v.length === 14) || 'Phone number is invalid'
-  ]
-
-  private emailMustMatch (): string {
-    return (this.contact.email === this.contact.confirmEmail) ? '' : 'Email addresses must match'
-  }
-
-  private get hasBusinessContactInfoChange () {
-    return !this.isSame(this.businessContact, this.getOriginalBusinessContact())
-  }
+import { AuthApiMixin, CommonMixin } from '@/mixins'
 
-  private updateContactInfo (): void {
-    this.emitContactInfo(this.contact)
-    this.isEditing = false
-    this.emitHaveChanges(this.hasBusinessContactInfoChange)
-  }
+// Components
+import { ContactInfo } from '@bcrs-shared-components/contact-info'
 
-  private resetContactInfo (): void {
-    this.contact = this.getOriginalBusinessContact()
-    this.emitContactInfo(this.contact)
-    this.emitHaveChanges(false)
-    this.isEditing = false
+@Component({
+  components: {
+    ContactInfo
   }
-
-  private getOriginalBusinessContact (): BusinessContactIF {
-    return {
-      email: this.originalBusinessContact.email,
-      confirmEmail: this.originalBusinessContact.email,
-      phone: this.originalBusinessContact.phone,
-      extension: this.originalBusinessContact.extension
-    }
+})
+export default class BusinessContactInfo extends Mixins(AuthApiMixin, CommonMixin) {
+  @Action setBusinessContact!: ActionBindingIF
+
+  @Getter getBusinessContact!: BusinessContactIF
+  @Getter getOriginalIA!: IncorporationFilingIF
+  @Getter getOriginalSnapshot!: BusinessSnapshotIF
+
+  /** Get the original Contact info dependant on filing type. */
+  private get originalContact (): BusinessContactIF {
+    return this.isCorrectionView()
+      ? this.getOriginalIA.incorporationApplication.contactPoint
+      : this.getOriginalSnapshot[5]
```

Comment:
I recommend a getter for this to abstract which array item this data is.

(ATM this requires too much knowledge of how snapshots work.)

Reply:
Fair point! Will add an getter a little more explicit to what we are `get`'ing here

Reply:
Updated in next commit 👍